opflexagent/utils/bridge_managers/ovs_lib.py

```diff
@@ -47,3 +51,10 @@ def ovsdb_transaction(self):
                     yield txn
                 finally:
                     self._transaction = None
+
+
+class FakeOVSBridge(OVSBridge):
+
+    def get_vif_port_by_id(self, port_id):
+        return ovs_lib.VifPort(port_id, None, port_id,
+            net.get_random_mac(cfg.CONF.base_mac.split(':')), self)
```

Comment:
every call to the same port ID will generate a new random mac. Is that what we want? Should we keep the generated value in memory if the same port is called twice?

Reply:
It ends up the MAC generated here won't get used. The fake manager only gets triggered by port update events, which means the MAC is already assigned at the plugin. The places where the VIF port object's MAC is referenced are also checking the details first:

https://github.com/noironetworks/python-opflex-agent/blob/master/opflexagent/utils/ep_managers/endpoint_file_manager.py#L114

https://github.com/noironetworks/python-opflex-agent/blob/master/opflexagent/utils/ep_managers/endpoint_file_manager.py#L156

https://github.com/noironetworks/python-opflex-agent/blob/master/opflexagent/utils/ep_managers/endpoint_file_manager.py#L262

Reply:
I guess I should try just passing in None as the MAC -- that should work as well.